test/functional/tests/io_class/test_io_class_stats_file_size_core.py

```diff
@@ -0,0 +1,218 @@
+#
+# Copyright(c) 2022 Intel Corporation
+# Copyright(c) 2024 Huawei Technologies Co., Ltd.
+# SPDX-License-Identifier: BSD-3-Clause
+#
+
+import re
+import pytest
+
+from datetime import timedelta
+from api.cas import casadm, ioclass_config
+from api.cas.cache_config import CacheMode, CleaningPolicy, SeqCutOffPolicy
+from api.cas.ioclass_config import IoClass
+from core.test_run_utils import TestRun
+from storage_devices.disk import DiskTypeSet, DiskType, DiskTypeLowerThan
+from test_tools import fs_utils
+from test_tools.disk_utils import Filesystem
+from test_tools.fio.fio import Fio
+from test_tools.fio.fio_param import IoEngine, ReadWrite
+from test_utils.os_utils import sync, drop_caches
+from test_utils.size import Size, Unit
+from tests.io_class.io_class_common import template_config_path
+
+
+@pytest.mark.parametrize("cache_mode", CacheMode)
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_io_class_stats_file_size_core_fs(cache_mode):
+    """
+    title: Open CAS statistics values per core for IO classification by file size.
+    description: |
+      Check Open CAS ability to assign correct IO class based on the file size.
+      Test checking configuration with different filesystems on each core.
+    pass_criteria:
+        - after FIO with direct statistics increases only for direct IO class
+    """
+
+    with TestRun.step("Prepare devices."):
+        cache_device, core_devices = prepare_devices(cores_count=len(Filesystem))
+
+    with TestRun.step("Start cache and add core devices (one for each filesystem)."):
+        cache, cores = prepare_cache_and_cores(cache_device, cache_mode, core_devices)
+
+    with TestRun.step("Make filesystem on OpenCAS devices and mount it."):
+        for core, fs in zip(cores, Filesystem):
+            mount_point = core.path.replace('/dev/', '/mnt/')
+            core.create_filesystem(fs)
+            core.mount(mount_point)
+
+    with TestRun.step("In default IO class configuration file set eviction priority=1 "
+                      "and load it"):
+        io_classes = prepare_io_classes(cache)
+        *file_size_based_io_classes, _ = io_classes
+
+    with TestRun.step("Run fio for all devices checking IO class statistics."):
+        sizes = prepare_fio_sizes(file_size_based_io_classes)
+        size_min = Size(512, Unit.Byte)
+
+        for io_class, size in zip(file_size_based_io_classes, sizes):
+            with TestRun.step(f"Testing IO class {io_class.id}."):
+                with TestRun.step("Run fio for each device and check that "
+                                  "only for tested IO class statistics increases."):
```

Comment:
Does this step nesting look ok in logs? Maybe TestRun.group() should be used?
For lines 59-60 - maybe TestRun.iteration()?